code/modules/reagents/Chemistry-Reagents.dm

```diff
@@ -8247,16 +8247,19 @@ var/global/list/tonio_doesnt_remove=list("tonio", "blood")
 	dupeable = FALSE
 
 	var/minimal_dosage = 1 //At least 1 unit is needed for petriication
+	var/is_being_petrified = FALSE
 
 /datum/reagent/petritricin/on_mob_life(var/mob/living/M)
 	if(..())
 		return 1
-
-	if(volume >= minimal_dosage && prob(30))
+	if(M.reagents.has_reagent(SACID) || M.reagents.has_reagent(PACID) || M.reagents.has_reagent(ACIDSPIT) || M.reagents.has_reagent(ACIDTEA))
```

Comment:
`has_any_reagents` is more efficient

Reply:
Sleeping inside `slow_pertify` is questionable, too. It should probably be handled by `Life()`?